### ISSUE TITLE
CA1508 False positive on Nullable DateTime

### ISSUE THREAD

**Issue by langdonx** [dotnet/roslyn-analyzers#4415]
### Analyzer

**Diagnostic ID**: [CA1508](https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1508): `Avoid dead conditional code`

### Analyzer source

**NuGet Package**: [Microsoft.CodeAnalysis.FxCopAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.FxCopAnalyzers)

**Version**: v3.3.1 (Latest)

### Describe the bug

It's reporting a false positive checking that `DateTime? != null`:

```
var punchIn = (DateTime?)t.GetProperty($"SchedulePunchIn{i}Time").GetValue(dbSchedule);
var punchOut = (DateTime?)t.GetProperty($"SchedulePunchOut{i}Time").GetValue(dbSchedule);

if (punchIn != null || punchOut != null) // this line here
```

### Steps To Reproduce

Simply write the following code:
```
var a = (DateTime?)DateTime.UtcNow;
if (a != null) { }
```

### Expected behavior

It shouldn't cause a warning.

### Actual behavior

It causes a warning.

## Additional context

N/A
